std/conv.d

```diff
@@ -3709,12 +3709,30 @@ private S textImpl(S, U...)(U args)
     {
         return null;
     }
+    else static if (U.length == 1)
+    {
+        return args[0].to!S;
+    }
     else
     {
-        auto result = to!S(args[0]);
-        foreach (arg; args[1 .. $])
-            result ~= to!S(arg);
-        return result;
+        import std.array : appender;
+
+        if (__ctfe)
+        {
+            auto result = to!S(args[0]);
+            foreach (arg; args[1 .. $])
+                result ~= to!S(arg);
+            return result;
+        }
+        else
+        {
+            auto app = appender!S();
+            app.reserve(U.length * S.sizeof);
+
+            foreach (arg; args)
+                app.put(to!S(arg));
```

Comment:
Still creating one string per argument. How about using `std.format.formatValue` to format directly into the appender?